src/app/reporting/Engine.h

```diff
@@ -119,6 +127,8 @@ class Engine
      *
      */
     uint32_t mCurReadHandlerIdx = 0;
+
+    ClusterInfo * mpChangePathList = nullptr;
```

Comment:
I also don't think it's a wise idea to have this global dirty rely on the same ClusterInfo resource pool that is used for allocating paths on incoming subscribe/read processing. This couples one onto the other - if there are a large number of subscribers who use up the majority of the pool, your dirty tracking system is not going to function well, if it all.

They have separate purposes, so having separate pools would be more sensible. Fine to do this as a follow-up issue.

Reply:
sure, would follow-up it to create the global dirty path list with object pool, https://github.com/project-chip/connectedhomeip/issues/9551